src/SFA.DAS.Commitments.Api.IntegrationTests/ApiHost/TestRegistry.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.IO;
+using System.Web;
+using System.Web.SessionState;
+using SFA.DAS.NLog.Logger;
+using StructureMap;
+
+namespace SFA.DAS.Commitments.Api.IntegrationTests.ApiHost
+{
+    public class TestRegistry : Registry
+    {
+        public TestRegistry()
+        {
+            // OWIN self-hosting doesn't have a HttpContext, but we need one because
+            // StructureMapDependencyScope uses the HttpContext to stash a NestedContainer
+            // if we get the authorization working, we could change StructureMapDependencyScope
+            // (also WebLoggingContext uses it, but we replace the one in the container with a null htppcontext - which it handles fine)
+            var httpContext = FakeHttpContext("http://localhost");
```

Comment:
I know currently we don't use the HttpContext in a per-request basis, but if we do, this will treat it as a singleton. Is it possible here to create a factory which would generate the FakeHttpContext per-request. (I don't know if this is possible, as I've not used StructureMap before).